checkstyle-tester/launch.groovy

```diff
@@ -0,0 +1,191 @@
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.util.regex.Pattern
+
+def cli = new CliBuilder(usage: 'launch [options]')
+cli.with {
+    h(longOpt: 'help', 'Show usage information', required: false)
+    c(longOpt: 'config', 'Checkstyle configuration file', args: 1, required: true)
+    p(longOpt: 'projectsToTestOn', 'File which contains defaultValuedefaultValueinfo about projects for testing', args: 1, required: true)
+}
+
+def options = cli.parse(args)
+if (areValidCliOptions(options)) {
+    if (options.h) {
+        cli.usage()
+        System.exit(0)
+    }
+    else {
+        def srcDir = 'src/main/java'
+        generateCheckstyleReport(options.config, srcDir, options.projectsToTestOn)
+    }
+}
+else {
+    println 'Error: invalid command line options!'
+    cli.usage()
+    System.exit(-1);
+}
+
+def areValidCliOptions(options) {
+    def valid = true
+    if (options) {
+        def fileWithProjectsInfo = new File(options.projectsToTestOn)
+        def configFile = new File(options.config)
+        if (!fileWithProjectsInfo.exists()) {
+            println "Error: file ${projectsToTestOn.getName()} does not exist!"
+            valid = false
+        }
+        if (!configFile.exists()) {
+            println "Error: file ${configFile.getName()} does not exist!"
+            valid = false
+        }
+    }
+    else {
+        valid = false
+    }
+    return valid
+}
+
+def generateCheckstyleReport(checkstyleConfig, srcDir, projectsToTestOn) {
+    println 'Testing Checkstyle started'
+    def projectsToTestOnFile = new File(projectsToTestOn)
+    def projects = projectsToTestOnFile.readLines()
+    def excludes = ""
+
+    def REPO_NAME_PARAM_NO = 0;
+    def REPO_TYPE_PARAM_NO = 1;
+    def REPO_URL_PARAM_NO = 2;
+    def REPO_COMMIT_ID_PARAM_NO = 3;
+    def REPO_EXCLUDES_PARAM_NO = 4;
+    def FULL_PARAM_LIST_SIZE = 5;
+
+    projects.each {
+        project ->
+            if (!project.startsWith('#') && !project.isEmpty()) {
+                def params = project.split('\\|')
+                def repoName = params[REPO_NAME_PARAM_NO]
+                def repoType = params[REPO_TYPE_PARAM_NO]
+                def repoUrl = params[REPO_URL_PARAM_NO]
+                def commitId = params[REPO_COMMIT_ID_PARAM_NO]
+                if (params.length == FULL_PARAM_LIST_SIZE) {
+                    excludes += ',' + params[REPO_EXCLUDES_PARAM_NO]
+                }
+                cloneRepository(repoName, repoType, repoUrl, commitId, srcDir)
+            }
+    }
+    runMavenExecution(srcDir, excludes, checkstyleConfig)
```

Comment:
please be consistent in sequence of arguments. I like this sequence , so please fix "+def generateCheckstyleReport(checkstyleConfig, srcDir, projectsToTestOn) {"
to be 
"def generateCheckstyleReport(srcDir, projectsToTestOn, checkstyleConfig) {"